app/routes/book_routes.py

```diff
@@ -1,7 +1,26 @@
-from flask import Blueprint, abort, make_response
+from flask import Blueprint, abort, make_response, request
+from app.models.book import Book
+from ..db import db
 
 books_bp = Blueprint("books_bp", __name__, url_prefix="/books")
 
+@books_bp.post("")
+def create_book():
+    request_body = request.get_json()
+    title = request_body.get("title")
+    description = request_body.get("description")
```

Comment:
👍

Reply:
Coming back to this after looking at the PUT route, `title` and `description` were effectively marked non-nullable due to how the model is set up. If either field is missing in the request body, `get` will silently return `None`, but then the null constraints will kick in at the commit below, raising `NotNullViolation`.

Later, when we refactor the creation logic into a `from_dict` we switch back to a direct key access, which lets us handle the KeyError in a helper. Could we stick with direct key access here, and if the text doesn't already mention it, we could do a callout that yes, this can crash if any required keys are missing, but that the reader could add some logic themselves for now if they wish, and we'll address it later on.

Reply:
I had been thinking about it as something to consciously refactor: moving up where the error happens if we know we'll get an error if those fields are missing, but we can still call out that difference of where errors would raise in the text and move it to direct key access here.

Reply:
In the grand scheme of things, I do like being aware that the database calls can throw and needing to potentially handle those. But given that we're switching over to a method that does direct key access, then minimizing the moving pieces feels easier to follow to me. We can still discuss that as written, there is an error condition. But with direct key access, it's a simple scenario that folks have experienced before rather than adding the complexity of the database exception into the mix. Even though, strictly speaking, we still shouldn't trust that every one of our database calls will succeed. 😓